From 19838a57c60008cbb51cd10e124347a198349706 Mon Sep 17 00:00:00 2001
From: Deepa G K <g.k.deepa@intel.com>
Date: Mon, 29 Feb 2016 15:53:50 +0530
Subject: [PATCH 3/3] Asoc: rt5645: Klockwork fix for Unvalidated integer value

Klockwork reported an issue where an Unvalidated integer value '*buf'
is received from 'memdup_user' and can be used to access array within the
same function rt5645_ioctl_common().

Tracked-On: https://jira01.devtools.intel.com/browse/OAM-21890
Change-Id: If046e873393ae2249d62817c301e0026029b545f
Signed-off-by: Deepa G K <g.k.deepa@intel.com>
Signed-off-by: Kishore Kadiyala <kishore.kadiyala@intel.com>
Reviewed-on: https://android.intel.com:443/475711
---
 sound/soc/codecs/rt5645_ioctl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/codecs/rt5645_ioctl.c b/sound/soc/codecs/rt5645_ioctl.c
index d71c26c..7bc8259 100644
--- a/sound/soc/codecs/rt5645_ioctl.c
+++ b/sound/soc/codecs/rt5645_ioctl.c
@@ -132,6 +132,8 @@ int rt5645_ioctl_common(struct snd_hwdep *hw, struct file *file,
 	case RT_SET_CODEC_HWEQ_IOCTL:
 		if (eq_mode[0] == *buf)
 			break;
+		if (*buf < 0 || *buf >= EQ_CH_NUM)
+			goto err;
 		eq_mode[*buf] = *(buf + 1);
 		rt5645_update_eqmode(codec, eq_mode[*buf], *buf);
 		break;
-- 
1.9.1

